src/rust/kafka/src/lib.rs

```diff
@@ -0,0 +1,279 @@
+use std::{
+    fmt::Display,
+    marker::PhantomData,
+};
+
+use futures::{
+    stream::{
+        Stream,
+        StreamExt,
+    },
+    FutureExt,
+    TryFutureExt,
+    TryStreamExt,
+};
+use rdkafka::{
+    config::ClientConfig,
+    consumer::stream_consumer::StreamConsumer,
+    error::KafkaError,
+    producer::{
+        FutureProducer,
+        FutureRecord,
+    },
+    util::Timeout,
+    Message,
+};
+use rust_proto_new::{
+    SerDe,
+    SerDeError,
+};
+use thiserror::Error;
+
+//
+// Kafka configurations
+//
+
+#[derive(Error, Debug)]
+pub enum ConfigurationError {
+    #[error("failed to retrieve value from environment variable {0}")]
+    EnvironmentError(#[from] std::env::VarError),
+
+    #[error("failed to construct kafka producer {0}")]
+    ProducerCreateFailed(KafkaError),
+
+    #[error("failed to construct kafka consumer {0}")]
+    ConsumerCreateFailed(KafkaError),
+
+    #[error("failed to subscribe kafka consumer {0}")]
+    SubscriptionFailed(KafkaError),
+}
+
+fn configure() -> Result<ClientConfig, std::env::VarError> {
+    let bootstrap_servers = std::env::var("KAFKA_BOOTSTRAP_SERVERS")?;
+    if bootstrap_servers.starts_with("SASL_SSL") {
```

Comment:
This is a comma separated list of URLs. Here I've probably assumed it's just _one_ URL because (so far) we only ever use one. I'll fix this here (and in the python code as well) to make sure we handle cases where more than one URL is provided.

Reply:
BTW it looks like that link I wrote in the comment is broken already 🙃. I'll try to find more permanent docs for these configuration parameters...

Reply:
Actually, there's no need to parse the comma-separated list. According to the rust-rdkafka docs this list is passed as a comma-separated string to the underlying librdkafka. Gotta love stringly typed interfaces!

https://docs.rs/rdkafka/latest/rdkafka/config/index.html#frequently-used-parameters